File: ui-android-view/android-views/src/main/java/com/ztiany/view/AppContext.java
package com.ztiany.view;

import android.app.Application;

import com.ztiany.view.utils.BaseUtils;

/**
 * @author Ztiany
 * Email: ztiany3@gmail.com
 * Date : 2017-10-16 22:59
 */
public class AppContext extends Application {

    @Override
    public void onCreate() {
        super.onCreate();
        BaseUtils.init(this);
    }

}

File: 05-architecture/Kotlin-with-Jetpack-Latest/app/src/main/java/com/ztiany/androidx/kotlin/AppContext.kt
package com.ztiany.androidx.kotlin

import android.content.Context
import androidx.appcompat.app.AppCompatDelegate
import androidx.multidex.MultiDexApplication

/**
 *@author Ztiany
 *      Email: ztiany3@gmail.com
 *      Date : 2017-07-09 18:51
 */
class AppContext : MultiDexApplication() {

    private object C {
        lateinit var instance: AppContext
    }

    override fun attachBaseContext(base: Context?) {
        super.attachBaseContext(base)
        C.instance = this
    }

    override fun onCreate() {
        super.onCreate()
        AppCompatDelegate.setCompatVectorFromResourcesEnabled(true)
    }

    companion object {
        fun getInstance(): AppContext {
            return C.instance
        }
    }

}